openshift_resources/oadp/data_protection_application.py

```diff
@@ -1,17 +1,516 @@
+        description='snapshotLocations defines the list of desired configuration to use for VolumeSnapshotLocations',
+    )
+    unsupportedOverrides: Optional[Dict[str, str]] = Field(
+        None,
+        description='unsupportedOverrides can be used to override images used in deployments. Available keys are:   - veleroImageFqin   - awsPluginImageFqin   - openshiftPluginImageFqin   - azurePluginImageFqin   - gcpPluginImageFqin   - csiPluginImageFqin   - dataMoverImageFqin   - resticRestoreImageFqin   - kubevirtPluginImageFqin',
+    )
+
+
+class DataProtectionApplicationConditionConditionStatus(Enum):
+    True_ = 'True'
+    False_ = 'False'
+    Unknown = 'Unknown'
+
+
+class DataProtectionApplicationCondition(BaseModel):
+    lastTransitionTime: datetime = Field(
+        ...,
+        description='lastTransitionTime is the last time the condition transitioned from one status to another. This should be when the underlying condition changed.  If that is not known, then using the time when the API field changed is acceptable.',
+    )
+    message: constr(max_length=32768) = Field(
+        ...,
+        description='message is a human readable message indicating details about the transition. This may be an empty string.',
+    )
+    observedGeneration: Optional[conint(ge=0)] = Field(
+        None,
+        description='observedGeneration represents the .metadata.generation that the condition was set based upon. For instance, if .metadata.generation is currently 12, but the .status.conditions[x].observedGeneration is 9, the condition is out of date with respect to the current state of the instance.',
+    )
+    reason: constr(
+        regex=r'^[A-Za-z]([A-Za-z0-9_,:]*[A-Za-z0-9_])?$', min_length=1, max_length=1024
+    ) = Field(
+        ...,
+        description="reason contains a programmatic identifier indicating the reason for the condition's last transition. Producers of specific condition types may define expected values and meanings for this field, and whether the values are considered a guaranteed API. The value should be a CamelCase string. This field may not be empty.",
+    )
+    status: DataProtectionApplicationConditionConditionStatus = Field(
+        ..., description='status of the condition, one of True, False, Unknown.'
+    )
+    type: constr(
+        regex=r'^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])$',
+        max_length=316,
+    ) = Field(
+        ...,
+        description='type of condition in CamelCase or in foo.example.com/CamelCase. --- Many .condition.type values are consistent across resources like Available, but because arbitrary conditions can be useful (see .node.status.conditions), the ability to deconflict is important. The regex it matches is (dns1123SubdomainFmt/)?(qualifiedNameFmt)',
+    )
+
+
+class DataProtectionApplicationStatus(BaseModel):
+    conditions: Optional[List[DataProtectionApplicationCondition]] = None
```

Comment:
please use https://github.com/oadp-qe/openshift-resources/blob/main/src/resources/io/k8s/apimachinery/pkg/apis/meta/v1.py#L332 as the elements' type of the list

Reply:
don't see you renamed it, so I unresolved the thread